From b2ad4a0d31083a5fc1dbfb65b51bff14b2b7942f Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Wed, 2 Sep 2020 10:45:48 -0700
Subject: [PATCH] Bug 1661378 - pkix: Do not use NULL where 0 is needed Clang
 finds this error

pkix_logger.c:316:32: error: cast to smaller integer type 'PKIX_ERRORCLASS' from 'void *' [-Werror,-Wvoid-pointer-to-enum-cast]
        logger->logComponent = (PKIX_ERRORCLASS)NULL;
                               ^~~~~~~~~~~~~~~~~~~~~
pkix_logger.c:617:32: error: cast to smaller integer type 'PKIX_ERRORCLASS' from 'void *' [-Werror,-Wvoid-pointer-to-enum-cast]
        logger->logComponent = (PKIX_ERRORCLASS)NULL;
                               ^~~~~~~~~~~~~~~~~~~~~
2 errors generated.

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 lib/nss/libpkix/pkix/util/pkix_logger.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/nss/lib/nss/libpkix/pkix/util/pkix_logger.c b/nss/lib/libpkix/pkix/util/pkix_logger.c
index a916e6e4f..10f537a31 100644
--- a/nss/lib/nss/libpkix/pkix/util/pkix_logger.c
+++ b/nss/lib/libpkix/pkix/util/pkix_logger.c
@@ -313,7 +313,7 @@ pkix_Logger_Destroy(
 
         logger->callback = NULL;
         PKIX_DECREF(logger->context);
-        logger->logComponent = (PKIX_ERRORCLASS)NULL;
+        logger->logComponent = (PKIX_ERRORCLASS)0;
 
 cleanup:
 
@@ -614,7 +614,7 @@ PKIX_Logger_Create(
 
         logger->callback = callback;
         logger->maxLevel = 0;
-        logger->logComponent = (PKIX_ERRORCLASS)NULL;
+        logger->logComponent = (PKIX_ERRORCLASS)0;
 
         PKIX_INCREF(loggerContext);
         logger->context = loggerContext;
-- 
2.41.0

